src/bm_sim/calculations.cpp

```diff
@@ -507,6 +582,37 @@ template class CustomCrcMgr<uint16_t>;
 template class CustomCrcMgr<uint32_t>;
 template class CustomCrcMgr<uint64_t>;
 
+namespace detail {
+
+std::ostream &operator<<(std::ostream &out, const rss_key_t &rss_key) {
+  out << "0x";
+  for (const auto& byte : rss_key) {
+    out << std::setfill('0') << std::setw(2) << std::hex << byte;
+  }
+  return out;
+}
+
+}  // namespace detail
+
+ToeplitzRssErrorCode
+ToeplitzRssMgr::update_key(NamedCalculation *calculation,
+                   const rss_key_t &key) {
+  // TODO(qobilidop): Make the logger work.
+  // Logger::get()->info("Updating RSS key of {} to: {}",
+  //                     calculation->get_name(), key);
```

Comment:
I got stuck on this. If I uncomment the logger code, it would result in an error saying:
```console
../../third_party/spdlog/bm/spdlog/sinks/./../details/./format.h:2567:6: error: no match for 'operator<<' (operand types are 'std::__cxx11::basic_ostringstream<char>' and 'const std::vector<unsigned char>')
 2567 |   os << value;
      |   ~~~^~~~~~~~
```

But `operator<<` is overloaded right above. I don't know what's wrong.

Reply:
I ended up using `ByteContainer` to replace `std::vector<uint8_t>`, which already has a `to_hex()` method defined.